appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/eesecurity/SignedJWTIdentityStore.java

```diff
@@ -214,8 +223,12 @@ public CredentialValidationResult validate(SignedJWTCredential signedJWTCredenti
             return Optional.empty();
         }
 
-        try (InputStream inputStream = publicKeyURL.openStream();
-             BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))){
+        URLConnection urlConnection = publicKeyURL.openConnection();
```

Comment:
You might set the Accept-Charset header before opening the connection thus asking for a Charset.

Reply:
Not needed at the moment